swayfmt/src/items/item_fn/mod.rs

```diff
@@ -26,15 +26,26 @@ impl Format for ItemFn {
         formatted_code: &mut FormattedCode,
         formatter: &mut Formatter,
     ) -> Result<(), FormatterError> {
-        self.fn_signature.format(formatted_code, formatter)?;
-        let body = self.body.get();
-        if !body.statements.is_empty() || body.final_expr_opt.is_some() {
-            Self::open_curly_brace(formatted_code, formatter)?;
-            self.body.get().format(formatted_code, formatter)?;
-            Self::close_curly_brace(formatted_code, formatter)?;
-        } else {
-            write!(formatted_code, " {{}}")?;
-        }
+        formatter.with_shape(
+            Shape::from(
+                &formatter.shape,
+                Some(0), // In some cases we will want to update parts of Shape
+                Some(CodeLine::new(LineStyle::Normal, ExprKind::Function)),
+            ),
+            |formatter| -> Result<(), FormatterError> {
+                self.fn_signature.format(formatted_code, formatter)?;
+                let body = self.body.get();
+                if !body.statements.is_empty() || body.final_expr_opt.is_some() {
+                    Self::open_curly_brace(formatted_code, formatter)?;
+                    self.body.get().format(formatted_code, formatter)?;
```

Comment:
Could this be `body.format(formatted_code, formatter)?;`?

Reply:
ah yep